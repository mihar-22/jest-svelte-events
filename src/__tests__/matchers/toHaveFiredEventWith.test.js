import '../../'

import { render, fireEvent } from '@testing-library/svelte'

import Comp from '../fixtures/Comp.svelte'

describe('.toHaveFiredEventWith', () => {
  test('passes when an event contains the correct payload', async () => {
    const { component, getByText } = render(Comp)
    listen(component, 'e2')
    const button = getByText('e2')
    await fireEvent.click(button)
    expect(component).toHaveFiredEventWith('e2', 50)
  })

  test('fails when an event fires without the correct payload', async () => {
    const { component, getByText } = render(Comp)
    listen(component, 'e2')
    const button = getByText('e2')
    await fireEvent.click(button)
    expect(() => {
      expect(component).toHaveFiredEventWith('e2', 100)
    }).toThrowErrorMatchingSnapshot()
  })

  test('does not print out received when event did not fire', async () => {
    const { component } = render(Comp)
    listen(component, 'e2')
    expect(() => {
      expect(component).toHaveFiredEventWith('e2', 100)
    }).toThrowErrorMatchingSnapshot()
  })

  test('prints out all received payloads on fail', async () => {
    const { component, getByText } = render(Comp)
    listen(component, 'e2')
    const button = getByText('e2')
    await fireEvent.click(button)
    await fireEvent.click(button)
    await fireEvent.click(button)
    expect(() => {
      expect(component).toHaveFiredEventWith('e2', 100)
    }).toThrowErrorMatchingSnapshot()
  })

  test('prints out objects correctly on fail', async () => {
    const { component, getByText } = render(Comp)
    listen(component, 'e4')
    const button = getByText('e4')
    await fireEvent.click(button)
    expect(() => {
      expect(component).toHaveFiredEventWith('e4', 100)
    }).toThrowErrorMatchingSnapshot()
  })
})

describe('.not.toHaveFiredEventWith', () => {
  test('passes when an event does not contain the correct payload', async () => {
    const { component } = render(Comp)
    listen(component, 'e2')
    expect(component).not.toHaveFiredEventWith('e2', 100)
  })

  test('fails when an event fires with the correct payload', async () => {
    const { component, getByText } = render(Comp)
    listen(component, 'e2')
    const button = getByText('e2')
    await fireEvent.click(button)
    expect(() => {
      expect(component).not.toHaveFiredEventWith('e2', 50)
    }).toThrowErrorMatchingSnapshot()
  })
})
