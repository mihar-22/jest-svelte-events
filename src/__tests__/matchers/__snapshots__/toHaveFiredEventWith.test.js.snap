// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`.not.toHaveFiredEventWith fails when an event fires with the correct payload 1`] = `
"[2mexpect([22m[31mcomponent[39m[2m).[22mnot[2m.[22mtoHaveFiredEventWith[2m([22m[32mexpected[39m[2m)[22m

Expected not to have fired with: [32m50[39m

Event: e2
Number of calls: [31m1[39m
Events stack: [31m[\\"e2\\"][39m"
`;

exports[`.toHaveFiredEventWith does not print out received when event did not fire 1`] = `
"[2mexpect([22m[31mcomponent[39m[2m).[22mtoHaveFiredEventWith[2m([22m[32mexpected[39m[2m)[22m

Expected: [32m100[39m

Event: e2
Number of calls: [31m0[39m
Events stack: [31m[][39m"
`;

exports[`.toHaveFiredEventWith fails when an event fires without the correct payload 1`] = `
"[2mexpect([22m[31mcomponent[39m[2m).[22mtoHaveFiredEventWith[2m([22m[32mexpected[39m[2m)[22m

Expected: [32m100[39m
Received: [31m50[39m

Event: e2
Number of calls: [31m1[39m
Events stack: [31m[\\"e2\\"][39m"
`;

exports[`.toHaveFiredEventWith prints out all received payloads on fail 1`] = `
"[2mexpect([22m[31mcomponent[39m[2m).[22mtoHaveFiredEventWith[2m([22m[32mexpected[39m[2m)[22m

Expected: [32m100[39m
Received: 
	   0: [31m50[39m,
	   1: [31m50[39m,
	   2: [31m50[39m

Event: e2
Number of calls: [31m3[39m
Events stack: [31m[\\"e2\\", \\"e2\\", \\"e2\\"][39m"
`;

exports[`.toHaveFiredEventWith prints out objects correctly on fail 1`] = `
"[2mexpect([22m[31mcomponent[39m[2m).[22mtoHaveFiredEventWith[2m([22m[32mexpected[39m[2m)[22m

Expected: [32m100[39m
Received: [31m{\\"a\\": 1, \\"b\\": 2}[39m

Event: e4
Number of calls: [31m1[39m
Events stack: [31m[\\"e4\\"][39m"
`;
